ARE/middleware/src/main/java/eu/asterics/mw/are/AsapiSupport.java

```diff
@@ -171,36 +175,6 @@ public AsapiSupport() {
     }
 
     /**
-     * Returns a formatted XML String of the componentType(s) in the bundle
-     * descriptor.
-     *
-     * @param bundleDescriptorURL
-     * @return
-     * @throws MalformedURLException
-     * @throws IOException
-     */
-    private String getFormattedBundleDescriptorStringOfComponentTypeId(URL bundleDescriptorURL)
-            throws MalformedURLException, IOException {
-        // Actually we should ask DefaultBundleModelParser to return just the
-        // part that belongs to the requested componentTypeId
-        // e.g. in case of AnalogIn there is also a second component type
-        // LegacyAnalogIn in the bundle_descriptor.xml
-        // Skip it for now because result is unformatted and JavaScript showed
-        // error callback.
-        // String
-        // bundleDescriptorString=DefaultBundleModelParser.instance.getBundleDescriptionOfComponentTypeId(componentTypeId,
-        // bundleDescriptorURI.toURL().openStream());
-
-        String bundle_descriptor = ResourceRegistry.getResourceContentAsString(bundleDescriptorURL.openStream());
-
-        bundle_descriptor = bundle_descriptor.replaceFirst("<\\?xml version=\"[0-9]\\.[0-9]\"\\?>", "");
-        bundle_descriptor = bundle_descriptor.replaceFirst("^(<componentTypes)?^[^>]*>", "");
-        bundle_descriptor = bundle_descriptor.replaceFirst("</componentTypes>", "");
-
-        return bundle_descriptor;
-    }
-
-    /**
      * Returns an xml String containing the component collection (bundle
      * descriptors of every AsTeRiCS component). This function searches in the
```

Comment:
"This function searches in the bin/ARE folder to discover the created components." is not correct anymore, is it? I think now we are just returning the cache file?!

Reply:
Yes, will fix this